src/providers/postgres/qgspostgresconn.h

```diff
@@ -183,6 +183,12 @@ class QgsPostgresConn : public QObject
     Q_OBJECT
 
   public:
+    /* @param shared allow using a shared connection. Should never be
+     *        called from a thread other than the main one. An
+     *        assertion guards against that.
+     *
+     *  @deprecated Use QgsPostgresConnPool
```

Comment:
there's no sip for this (and that's ok like that)

the reference to sip was more thought of as a general advice for deprecations in other places.